enhancements/bundle-validation.md

```diff
@@ -0,0 +1,220 @@
+
+## Motivation
+
+The primary motivation for this enhancement is to provide a library that statically validates a [bundle](https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-bundle.md). This library can be used by external projects and tools, such as [Operator-SDK](https://github.com/operator-framework/operator-sdk), to perform a static check for the validity of a bundle before testing the operator on clusters.
+
+Providing users with the tools to statically analyze bundles is a hard requirement from people like operator developers and pipeline owners due to the high cost of testing bundles on fresh clusters. Even though there is no replacement solution to testing operators on a running cluster, we would still like to provide static validations to help catch as many issues as possible. Currently, they rely on [operator-courier](https://github.com/operator-framework/operator-courier), a python project, for static analysis of bundle data. There are primarily two major shortcomings that exist with the operator-courier project:
+
+- **Issue 1:** The fact that the objects validated by the library are defined in OLM and written in Golang mismatches with the courier project language. The structs defined in go are updated frequently, causing courier to be out-of-sync and in need of constant updates.
+
+- **Issue 2:**  Courier was designed to [interact with App-Registries](https://github.com/operator-framework/operator-courier/blob/master/docs/how-to-upload-artifact-deprecated.md), which are being replaced by [Catalog Images](https://docs.openshift.com/container-platform/4.3/operators/olm-restricted-networks.html#olm-understanding-operator-catalog-images_olm-restricted-networks).
+
+The new validation library can address each of the aforementioned shortcomings by the following:
+
+- **Issue 1:** The library will be written in Golang and maintained as part of [operator-framework/api](https://github.com/operator-framework/api). The bundle structs will be moved to _operator-framework/api_ and act as the source of truth for the structure of these objects by the [OLM](https://github.com/operator-framework/operator-lifecycle-manager), [Operator-SDK](https://github.com/operator-framework/operator-sdk), and this validation library. This change will synchronize the state of the bundle structure across all projects making it easier to maintain, import, and serialize.
+
+- **Issue 2:** Rather than rewriting the courier project to provide static analysis for bundles provided as Catalog Images, the new library can be written and then imported by tools like _scorecard_ to perform validation.
+
+### Goals
+
+Define a set of rules/validations so that bundles with their specific set of configurations can be validated by checking the required information and components.
```

Comment:
Should be a list.

Reply:
Do you think what is in implementation detail is going to suffice?

Reply:
nvm... i see you meant format .... got it ...